src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/LoadProfilerActionTests.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Monitoring.TestCommon.Options;
+using Microsoft.Diagnostics.Monitoring.Tool.UnitTests.CollectionRules.Actions;
+using Microsoft.Diagnostics.Tools.Monitor;
+using Microsoft.Diagnostics.Tools.Monitor.CollectionRules;
+using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Actions;
+using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Options;
+using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Options.Actions;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.Diagnostics.Monitoring.Tool.UnitTests
+{
+    public sealed class LoadProfilerActionTests
+    {
+        private readonly ITestOutputHelper _outputHelper;
+        private static readonly TimeSpan TimeoutMs = TimeSpan.FromSeconds(1);
+        private const string DefaultRuleName = nameof(ActionDependencyAnalyzerTests);
+
+        public LoadProfilerActionTests(ITestOutputHelper outputHelper)
+        {
+            _outputHelper = outputHelper;
+        }
+
+        [Fact]
+        public async Task TranslateSingleStringToAny()
+        {
+            const string TargetPath = @"C:\My\Test\Path";
+            LogRecord record = new LogRecord();
+            LoadProfilerOptions settings = new LoadProfilerOptions()
+            {
+                Path = TargetPath,
+                ProfilerGuid = Guid.NewGuid(),                    
+            };
+            await TestHostHelper.CreateCollectionRulesHost(_outputHelper, rootOptions =>
+            {
+                CollectionRuleOptions options = rootOptions.CreateCollectionRule(DefaultRuleName)
+                    .AddAction(KnownCollectionRuleActions.LoadProfiler, settings, out CollectionRuleActionOptions actionOptions)
+                    .SetStartupTrigger();
+
+                settings = (LoadProfilerOptions)options.Actions.Last().Settings;
+            }, host =>
+            {
+                using CancellationTokenSource cancellationTokenSource = new CancellationTokenSource(TimeoutMs);
+
+                IOptionsMonitor<CollectionRuleOptions> collectionRuleOptsMonitor = host.Services.GetRequiredService<IOptionsMonitor<CollectionRuleOptions>>();
+                CollectionRuleOptions ruleOptions = collectionRuleOptsMonitor.Get(DefaultRuleName);
+                ILogger<CollectionRuleService> logger = host.Services.GetRequiredService<ILogger<CollectionRuleService>>();
```

Comment:
The `logger` and `context` are not really used/necessary.

Reply:
Nope, didn't realize `ruleOptions` was a post-configuration value.